ec2.tf

```diff
@@ -0,0 +1,181 @@
+locals {
+  ec2_capacity_providers_cleanup = { for name, provider in var.capacity_providers_ec2 :
+    name => { for key, value in provider : key => value if value != null }
+  }
+
+  ec2_capacity_provider_default = {
```

Comment:
why provide defaults here rather than within the optional() block for the var in variables.tf ?